### PR TITLE
[Tock Studio] Angular version upgrade (v14 to v17)

### PR DESCRIPTION
**Pull request by rkuffer**
Problems encountered:
- Temporary workaround must have been added for https://github.com/akveo/nebular/issues/3258
- Temporary workaround must have been added for https://github.com/akveo/nebular/issues/3241
- The display of the Sankey graph is no longer working (angular-google-charts dependency not maintained anymore and incompatible with Angular 15) and has therefore been disabled. A complete replacement of the Cytoscape graphs by their Apache Echarts equivalent is planned soon (including a Sankey display). Cf https://github.com/theopenconversationkit/tock/issues/1714
- After update, a last moderate security alert remains for Boostrap 4. A version upgrade to Bootstrap 5 is planned

### CODE REVIEW

**Review by AurelienSylvan — APPROVED** [review]

File: bot/admin/web/src/app/analytics/flow/flow.component.html [hunk 1]
```diff
@@ -466,7 +466,7 @@
 <div *ngIf="selectedLayout === 'Sankey'">
   <nb-card>
     <nb-card-body>
-      <google-chart
+      <!-- <google-chart
```
Comment (Benvii) [inline bot/admin/web/src/app/analytics/flow/flow.component.html:469]: Can create an issue on tock repository to follow the change with  Apache Echarts  graphs and mention it (the issue number) as a comment here.

File: bot/admin/web/src/app/analytics/flow/flow.component.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export class FlowComponent implements OnInit, OnDestroy {
   flowData: ChartData;
   loading: boolean = false;
 
-  SankeyChartType: ChartType = ChartType.Sankey;
+  // SankeyChartType: ChartType = ChartType.Sankey;
```
Comment (Benvii) [inline bot/admin/web/src/app/analytics/flow/flow.component.ts:163]: Can create an issue on tock repository to follow the change with  Apache Echarts  graphs and mention it (the issue number) as a comment here.

File: bot/admin/web/src/app/applications/applications.module.ts [hunk 3]
```diff
@@ -84,7 +82,6 @@ export class NlpApplicationConfig implements ApplicationConfig {
     NbInputModule,
     NgJsonEditorModule,
     NbDialogModule.forRoot(),
-    NgbPaginationModule,
```
Comment (Benvii) [inline bot/admin/web/src/app/applications/applications.module.ts:87]: Is this module replaced by an other pagination module ?

**Review by Benvii — APPROVED** [review]
Ok for me, just one comment about the chart component.

File: bot/admin/web/src/app/applications/applications.module.ts (hunk 3 above)
Comment (rkuffer) [reply bot/admin/web/src/app/applications/applications.module.ts:87]: Yes, on the bot module we have been using a homemade one for several months. This was the last component that forbade us to drop the ng-bootstrap dependency (we use bootstrap but only the css part of it)

File: bot/admin/web/src/app/analytics/flow/flow.component.html (hunk 1 above)
Comment (rkuffer) [reply bot/admin/web/src/app/analytics/flow/flow.component.html:469]: https://github.com/theopenconversationkit/tock/issues/1714

File: bot/admin/web/src/app/analytics/flow/flow.component.ts (hunk 2 above)
Comment (rkuffer) [reply bot/admin/web/src/app/analytics/flow/flow.component.ts:163]: https://github.com/theopenconversationkit/tock/issues/1714

**Pull request merged**